src/passes/PostAssemblyScript.cpp

```diff
@@ -364,8 +364,16 @@ struct OptimizeARC : public WalkerPass<PostWalker<OptimizeARC>> {
     return (*releaseLocation)->cast<Call>()->operands[0]->cast<LocalGet>();
   }
 
+  // Tests if the given retain is generally eliminable (does not retain an
+  // allocation and does not reach an escape)
+  bool testRetainEliminable(LocalSet* retain, AliasGraph& graph) {
```

Comment:
can the AliasGraph here be `const AliasGraph&`?

Reply:
Gave this a try, including updating the called functions herein, but the compiler then barks at me on `[]` accesses (no such operator) like `graph.setInfluences[...]` and `graph.getSetses[...]` in the called functions, as if `std::map` etc. can't be used this way. Might be missing something, as usual :)

Reply:
I think the issue is that `map[key]` can have side effects, it doesn't just do a read, but it also creates the entry (with a default value) if it isn't there.

You can do `map.at(key)` if you know the key must be there. that will not change the map, and it will also check it actually exists, so it's better (but again, only if you know the key must be there).